5.analyze-weights/gsea.py

```diff
@@ -0,0 +1,117 @@
+"""
+2018 Gregory Way
+5.analyze-components/gsea.py
+
+Submit jobs that perform GSEA on all compiled weight matrices
```

Comment:
it's interesting how difficult to it is to figure out which of the gsea-associated methods gseapy implemented. do you know which statistic it uses?

Reply:
yeah, it is interesting... I did a bit of digging:

1. I am running [`gsea.prerank`](https://github.com/zqfang/GSEApy/blob/c43a73ee638e4326e8d5505b92f9b6dfeb4d9400/gseapy/gsea.py#L986), which does not take in a `method` argument.
2. `gsea.prerank` instantiates a `Prerank` class, which also does not take in a `method` argument. It does call [`gsea_compute`](https://github.com/zqfang/GSEApy/blob/c43a73ee638e4326e8d5505b92f9b6dfeb4d9400/gseapy/gsea.py#L491), which, as a default, takes in a method argument of `None`.
3. [`gsea_compute`](https://github.com/zqfang/GSEApy/blob/e3f18b1ec1829b44a445e22fc4b6fe78c23f0f0a/gseapy/algorithm.py#L423) uses a `method` argument, but only when the `permutation_type` == "phenotype". Since the permutation type is "gene_set" by [default](https://github.com/zqfang/GSEApy/blob/c43a73ee638e4326e8d5505b92f9b6dfeb4d9400/gseapy/gsea.py#L493), the gsea compute command does not require a method. 

I suppose this is because the gene set is already preranked.

There is another tunable method called `weighted_score_type`, which [defaults to 1](https://github.com/zqfang/GSEApy/blob/c43a73ee638e4326e8d5505b92f9b6dfeb4d9400/gseapy/gsea.py#L987).

There is [some documentation](https://github.com/zqfang/GSEApy/blob/e3f18b1ec1829b44a445e22fc4b6fe78c23f0f0a/gseapy/algorithm.py#L17-L24) about this metric, but I think I will keep it as its default.